### DISCUSSION TITLE
Runtime Constants & Static Variables

### DISCUSSION THREAD

**Discussion by apainintheneck** [gleam-lang/gleam#3182] [Questions & support] [answered]
In other languages, there is the concept of runtime constants or static variables which can get initialized once and then just sit there for the life of the program. I was wondering if this was possible in Gleam.

I'm working my way through the Exercism course right now and I started reaching for that behavior when working with regexes. Normally in other languages I'd cache the compiled regex in a constant to make sure that it doesn't get recreated each time it gets used. The only alternatives I've been able to find is either creating it early on in the program and passing it around or caching it in some other instantiated type which will end up using it from time to time.

**Comment by lpil** [comment, marked as answer]
Gleam does not have any of those features, no.

You could choose to pass it in as an argument or you could use functionality such as persistent term or ETS to store it in a way that you can access non-locally.

**Reply by apainintheneck** [reply]
Good to know, thanks! I wasn't aware of persistent term and hadn't looked into ETS yet either. Persistent term seems like it would be a good fit here.

**Reply by apainintheneck** [reply]
Ah, I see this has already been mentioned before in the annoyances doc: https://github.com/gleam-lang/gleam/blob/main/docs/annoyances.md#cannot-shift-repeated-work-to-compile-or-initialise-time
